src/core/bank.ts

```diff
@@ -25,17 +26,33 @@ type ReactorConstructor<T extends Reactor, S> = {
 // console.log(inst.x)
 // console.log(inst.y)
 
-export class Bank<T extends Reactor,S> {
+export class Bank<T extends Reactor, S> extends Component {
+    private runtime!: Runtime;
+
+    public _receiveRuntimeObject(runtime: Runtime): void {
+        if (!this.runtime) {
+            this.runtime = runtime
+        } else {
+            throw new Error("Can only establish link to runtime once. Name: " + this._getFullyQualifiedName())
+        }
+    }
 	private readonly members: Array<T> = new Array();
-	constructor(width: number, cls:ReactorConstructor<T,S>, ...args:ReactorArgs<S>) {
+	constructor(container: Reactor, width: number, cls:ReactorConstructor<T, S>, bankIndexArgIndex?: number, ...args:ReactorArgs<S>) {
+        super(container)
 		
 		for (let i=0; i < width; i++) {
+            if (bankIndexArgIndex !== undefined) {
```

Comment:
This seems rather hacky and would be easy to get wrong when you're instantiating a bank manually...